schema.md

```diff
@@ -1,82 +1,71 @@
 
-/
-All data relating to a mcmc run.
+Currently there are 2 implementations of this design:
+* [ArviZ](https://arviz-devs.github.io/arviz/) in Python which integrates with:
+  - [emcee](https://emcee.readthedocs.io/en/stable/)
+  - [PyMC3](https://docs.pymc.io)
+  - [pyro](https://pyro.ai/)
+      and [numpyro](https://pyro.ai/numpyro/)
+  - [PyStan](https://pystan.readthedocs.io/en/latest/index.html),
+      [CmdStan](https://mc-stan.org/users/interfaces/cmdstan)
+      and [CmdStanPy](https://cmdstanpy.readthedocs.io/en/latest/index.html)
+  - [tensorflow-probability](https://www.tensorflow.org/probability)
+* [ArviZ.jl](https://github.com/sethaxen/ArviZ.jl) in julia which integrates with:
+  - [Turing](https://turing.ml/dev/).
 
-attrs:
-backend: "stan" or "pymc3"
-version: str
-model_name?: str
-comment?: str
-model_version?: str
-timestamp: int
-author?: str
+## Current design
+`InferenceData` stores all quantites relevant in order to fulfill its goals in different groups. Each group, described below, stores a conceptually different quantity generally represented by several multidimensional labeled variables.
 
-/coords
-For each dimension, we store a corresponding variable here, that contains the labels for that dimension.
+Each group should have one entry per variable, with the first two dimensions of each variable should be the sample identifier (`chain`, `draw`). Dimensions must be named and explicit their index values, called coordinates. Coordinates can have repeated identifiers and may not be numerical. Variable names must not share names with dimensions.
 
-/model
-Each backend can store a representation of the model in here. (I guess
-source code for stan, no clue for pymc3 at this point.)
+Moreover, each group contains the following attributes:
+* `created_at`: the date of creation of the group.
+* `inference_library`: the library used to run the inference.
+* `inference_library_version`: version of the inference library used.
 
-/sample_stats
-Statistics computed while sampling, like step size (step_size), depth, diverging, energy (for HMC), log probability (lp), and log likelihood (log_likelihood).
+`InferenceData` data objects contain any combination the groups described below.
 
-/initial_point
-    One point in parameter space for each chain, where that chain started.
-    TODO We could also store that as an attribute to each var in /trace
+#### `posterior`
+Samples from the posterior distribution p(theta|y).
 
-/tuning_trace?
-    Same as /trace, but optional. It contains the trace during tuning. Optional. (only if store_tune=True or so?)
+#### `sample_stats`
+Information and diagnostics about each `posterior` sample, provided by the inference backend. It may vary depending on the algorithm used by the backend (i.e. an affine invariant sampler has no energy associated). The name convention used for `sample_stats` variables is the following:
+* `lp`: unnormalized log probability of the sample
```

Comment:
`(unnormalized) joint log probability of the sample`

Also, is it `for sample` or `for model`

Reply:
I think it is for sample and model?

Reply:
It seems odd to me that the information in the Arviz `sample_stats` does not contain the information in `pymc3.stats.summary()`, which is variable-wise. ,instead of just sample-wise.

Shouldn't we pull that information into `sample_stats`, as well?

Reply:
In ArviZ.jl, I'm putting Turing's summary information into the attributes section, but if there's a better place for it, I'll put it there.